edxadapt/static/js/src/edxadapt.js

```diff
@@ -43,13 +44,17 @@ function EdxAdaptXBlock(runtime, element) {
             })
             .done(function() {
                 console.log("User Configured");
-                setStatusOk();
             })
             .fail(function() {
                 console.log('Failed to configure skill "'+skill+'" for user "'+studentId+'"');
+                ready_to_enroll = false;
                 setStatusNok();
             });
         });
+        if (ready_to_enroll) {
```

Comment:
Instead, you can make `configureUser()` return a list of deferreds and pass them in `registerUserInEdxAdapt()` using `$.when().then()` syntax. This way you will ensure `registerUser()` is called after all skills are configured.